tests/20-charm-validation.py

```diff
@@ -90,6 +90,19 @@ def test_worker_services(self):
             for service in services:
                 self.assertTrue(check_systemd_service(worker, service))
 
+    def test_scale_master_services(self):
+        '''Test we can scale kubernetes masters.'''
+        for master in self.masters:
+            self.deployment.remove_unit(master.info['unit_name'])
+        self.deployment.sentry.wait()
+        assert len(self.masters) is 0
+
+        self.deployment.add_unit('kubernetes-master', 2)
+        self.deployment.sentry.wait()
+        for master in self.masters:
+            output, rc = master.run('grep server: /home/ubuntu/config')
+            self.assertTrue(rc == 0)
```

Comment:
The server IP on the configs should be the IP of the loadbalancer. And you are right that we should make sure the loadbalancer points to the new masters as soon as they come up.

The latest commit does both checks. Thank you.